src/main/java/bruh/tasklist/TaskList.java

```diff
@@ -0,0 +1,128 @@
+package bruh.tasklist;
+
+import bruh.task.LocalDateTimeOrString;
+import bruh.task.Task;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Encapsulates a list of tasks, as well as associated functions.
+ */
+public class TaskList implements Serializable {
+    private final List<Task> tasks;
+    private final boolean isFiltered;
+
+    /**
+     * Constructor for a pre-existing task list.
+     *
+     * @param tasks The pre-existing list of tasks.
+     */
+    private TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+        this.isFiltered = true;
+    }
+
+    /**
+     * Constructor for a new task list.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+        this.isFiltered = false;
+    }
+
+
+    /**
+     * Returns a list of all the tasks in the list as a string.
+     *
+     * @return A string of all the tasks in the list.
+     */
+    public String listTasks() {
+        if (tasks.isEmpty()) {
+            return getTaskCountDesc();
+        }
+        StringBuilder sb = new StringBuilder();
+        sb.append("Here are the ").append(isFiltered ? "matching " : "").append("tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            sb.append(String.format("\n%d.%s", i + 1, tasks.get(i).toString()));
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Returns a description of the number of tasks in the list.
+     *
+     * @return A description of the number of tasks in the list.
+     */
+    public String getTaskCountDesc() {
+        return tasks.isEmpty() ? "There are no " + (isFiltered ? "matching " : "") + "tasks."
```

Comment:
I believe it would be better to separate this into a few lines to avoid confusion!:smile: For the conditional expressions, maybe you can put it in a separate variable such as 
String plurality = (tasks.size() == 1) ? "" : "s";

Reply:
Thanks for the suggestion! Agreed, it would be better to split this into multiple lines for more readability. Will fix!